cellarium/cas/data_preparation.py

```diff
@@ -97,11 +111,26 @@ def sanitize(
 
     # Create `obs` index
     obs = adata.obs.copy()
-    obs.index = pd.Index([str(x) for x in np.arange(adata.shape[0])])
-
+    obs.index = original_obs_ids
+
+    var_df_data = {}
+    if feature_names_column_name is not None:
+        adata_feature_name_list = _get_adata_var_index_or_by_column(
+            adata=adata, var_column_name=feature_names_column_name
+        )
+        gene_id_to_gene_symbol_map = {
+            gene_id: gene_symbol for gene_symbol, gene_id in zip(adata_feature_name_list, adata_feature_schema_list)
+        }
+        cas_feature_name_list = [
+            gene_id_to_gene_symbol_map[gene_id] if gene_id in gene_id_to_gene_symbol_map else "N/A"
```

Comment:
Let's log a warning that the gene symbol for gene_id was not available and it was replaced with "N/A". This issue also relates to one of the 10x asks, i.e. mainting a central ENSEMBL ID to Gene Symbol map (perhaps as a part of the schema). If we're getting the map from the server, then we shouldn't run into a "N/A" case at all. At this point, we're only keeping and iterating over gene IDs that are in the schema.

Reply:
Not exactly. Our schema is made based on the 10x schema, there are some genes in the schema that never expressed in our DB.

Reply:
Added logging a warning.